src/NLog/Common/InternalLogger.cs

```diff
@@ -246,40 +246,55 @@ private static void Write([CanBeNull]Exception ex, LogLevel level, string messag
                 }
                 var msg = builder.ToString();
 
-                // log to file
-                var logFile = LogFile;
-                if (!string.IsNullOrEmpty(logFile))
+                try
                 {
-                    using (var textWriter = File.AppendText(logFile))
+                    // log to file
+                    var logFile = LogFile;
+                    if (!string.IsNullOrEmpty(logFile))
                     {
-                        textWriter.WriteLine(msg);
+                        lock (LockObject)
+                        {
+                            using (var textWriter = File.AppendText(logFile))
+                            {
+                                textWriter.WriteLine(msg);
+                            }
+                        }
                     }
                 }
-
-                // log to LogWriter
-                var writer = LogWriter;
-                if (writer != null)
+                finally
                 {
-                    lock (LockObject)
+                    // log to LogWriter
+                    var writer = LogWriter;
```

Comment:
why is this code in the `finally`?  Is this really by-design? 

I only use `finally` to clean up stuff, never to do functional logic. Am I missing here something?